lib/stats/stats-prometheus.c

```diff
@@ -95,23 +95,34 @@ stats_format_prometheus_format_value(const StatsClusterKey *key, StatsCounterIte
       converted_int *= 60;
     case SCU_MINUTES:
       converted_int *= 60;
-      g_string_printf(value, "%"G_GUINT64_FORMAT, converted_int);
+    case SCU_SECONDS:
+      if (key->formatting.frame_of_reference == SCFOR_RELATIVE_TO_TIME_OF_QUERY)
+        {
+          unix_time_set_now(&now);
+          converted_double = (now.ut_sec + now.ut_usec / 1e6) - converted_int;
```

Comment:
this still does not fix the y2038 problem pointed out by @MrAnno 

I am not sure we need the resolution that floating point gives us here, especially as the timestamp is only updated once per second at most.

Reply:
You are right, I have removed the support for milli and nanosec.

I believe I have fixed the Y2038 problem, too. I hope I did not miss anything.